apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/match/HierarchyMatch.java

```diff
@@ -95,7 +95,7 @@ private void matchHierarchyClass(TypeDescription.Generic clazz, List<String> par
         }
 
         TypeDescription.Generic superClazz = clazz.getSuperClass();
-        if (superClazz != null && !clazz.getTypeName().equals("java.lang.Object")) {
+        if (superClazz != null && !"java.lang.Object".equals(clazz.getTypeName())) {
```

Comment:
When class type name is NULL?

Reply:
in this case, the class type is not null, but  equqls() was usually used in this style